packages/cli/src/commands/source/push/add.ts

```diff
@@ -65,38 +73,51 @@ export default class SourcePushAdd extends Command {
   public async run() {
     await this.showDeprecatedFlagWarning();
     const {args, flags} = await this.parse(SourcePushAdd);
-    const {accessToken, organization, environment, region} =
-      await new AuthenticatedClient().cfg.get();
-    const source = new PushSource(accessToken!, organization, {
-      environment,
-      region,
-    });
+    const source = await this.getSource();
 
     CliUx.ux.action.start('Processing files');
 
     const fileNames = await getFileNames(flags);
-    const options = {
+    const options: BatchUpdateDocumentsFromFiles = {
       maxConcurrent: flags.maxConcurrent,
       createFields: flags.createMissingFields,
+      fieldNameTransformer: flags.normalizeInvalidFields
+        ? BuiltInTransformers.toLowerCase
+        : BuiltInTransformers.identity,
     };
-    await source.setSourceStatus(args.sourceId, 'REFRESH');
 
+    await source.setSourceStatus(args.sourceId, 'REFRESH');
     await source
       .batchUpdateDocumentsFromFiles(args.sourceId, fileNames, options)
       .onBatchUpload((data) => this.successMessageOnAdd(data))
       .onBatchError((data) => this.errorMessageOnAdd(data))
       .batch();
 
-    await source.setSourceStatus(args.sourceId, 'IDLE');
-
     CliUx.ux.action.stop(green('✔'));
   }
 
   @Trackable()
   public async catch(err?: Error & {exitCode?: number}) {
+    formatErrorMessage(err);
+    CliUx.ux.action.stop(red.bold('!'));
     throw err;
   }
 
+  public async finally() {
+    const {args} = await this.parse(SourcePushAdd);
+    const source = await this.getSource();
+    await source.setSourceStatus(args.sourceId, 'IDLE');
```

Comment:
You should run the finally if and only if we did successfully affected the status of the source IMO.

Reply:
I understand, but this is a bit overkill I think.
If the first call of `setSourceStatus` returned a 2** status code, it is safe to assume that the second call will work as well.
If the first `setSourceStatus` call fails for some reason, the second will also fail.

Also since the API call responsable to change the status returns `null` as the response, it will be a bit too much to really know wether the CLI did actually change the source status.
We would need 2 API calls to fetch the source status (one at the very beginning and one after the status change).